generator/sbpg/targets/rust.py

```diff
@@ -60,6 +60,11 @@
 BASE_TIME_MSGS = ["MSG_OBS", "MSG_OSR", "MSG_SSR"]
 
 CUSTOM_GPS_TIME_MSGS = {
+    "MSG_GPS_TIME": """
+if !matches!(self.time_source(), Ok(TimeSource::GnssSolution) | Ok(TimeSource::Propagated)) {
+    return None;
+}
+""".strip() + GPS_TIME,
```

Comment:
code golf but could be:
```suggestion
    "MSG_GPS_TIME": """
self.time_source().ok()?
""".strip() + GPS_TIME,
```

Reply:
Still need to check that it's not equal to `TimeSource::None` though (which obviously is not very clear by the way I'm doing it now). Maybe

```rust
if self.time_source().ok()? == TimeSource::None {
    return None;
}
```
?

Reply:
i see, in that case i probably prefer it as written, but if we are using a rust version later than 1.70
```rust
if !self.time_source().is_ok_and(|x| x != TimeSource::None) {
     return None
}
```

but i think it is better as written.

Reply:
`is_ok_and` TIL! But yeah agreed it's probably fine as is